src/org/opensolaris/opengrok/index/PendingFileCompleter.java

```diff
@@ -350,7 +350,20 @@ private void doLink(PendingSymlinkageExec lnk) throws IOException {
             }
             Path sourcePath = Paths.get(lnk.source);
             deleteFileOrDirectory(sourcePath);
-            Files.createSymbolicLink(sourcePath, Paths.get(lnk.targetRel));
+
+            File sourceParentFile = sourcePath.getParent().toFile();
+            /**
+             * The double check-exists in the following conditional is necessary
+             * because during a race when two threads are simultaneously linking
+             * for a not-yet-existent `sourceParentFile`, the first check-exists
+             * will be false for both threads, but then only one will see true
+             * from mkdirs -- so the other needs a fallback again to
+             * check-exists.
+             */
+            if (sourceParentFile.exists() || sourceParentFile.mkdirs() ||
```

Comment:
Is there perhaps another solution ? What about making this code block synchronized ?